pkg/vpcmodel/drawioGenerator.go

```diff
@@ -54,3 +55,30 @@ func (gen *DrawioGenerator) TreeNode(res DrawioResourceIntf) drawio.TreeNodeInte
 func (exn *ExternalNetwork) GenerateDrawioTreeNode(gen *DrawioGenerator) drawio.TreeNodeInterface {
 	return drawio.NewInternetTreeNode(gen.PublicNetwork(), exn.CidrStr)
 }
+
+func (g *groupedEndpointsElems) GenerateDrawioTreeNode(gen *DrawioGenerator) drawio.TreeNodeInterface {
+	// todo - need to implement, currently blocked in parse_args.go:
+	return gen.TreeNode((*g)[0])
+}
+
+func (g *groupedExternalNodes) GenerateDrawioTreeNode(gen *DrawioGenerator) drawio.TreeNodeInterface {
+	if len(*g) == 1 {
+		return gen.TreeNode((*g)[0])
+	}
+	tooltip := []string{}
+	for _, n := range *g {
+		tooltip = append(tooltip, n.(*ExternalNetwork).Cidr())
+	}
+	name := "Various IP ranges"
+	if all, _ := isEntirePublicInternetRange(*g); all {
+		name = publicInternetNodeName
+	}
+	tn := drawio.NewInternetTreeNode(gen.PublicNetwork(), name)
+	tn.SetTooltip(tooltip)
+	return tn
+}
+
+func (n *VPCResource) IsExternal() bool           { return false }
+func (exn *ExternalNetwork) IsExternal() bool     { return true }
+func (g *groupedEndpointsElems) IsExternal() bool { return false }
+func (g *groupedExternalNodes) IsExternal() bool  { return true }
```

Comment:
Move these methods to the places where these types and their methods are defined

Reply:
same

Reply:
done